### PR TITLE
[FIXED JENKINS-47074] Support build ID in SpecificBuildSelector

### PR DESCRIPTION
**Pull request by ikedam**
https://issues.jenkins-ci.org/browse/JENKINS-47074

It makes it easier to use SpecificBuildSelector in pipeline jobs like this:
* Before
    ```
    copyArtifacts(projectName: "fromJob", selector: specific("${build("fromJob").number}")
    ```
* After
    ```
    copyArtifacts(projectName: "fromJob", selector: specific("build("fromJob").id")
    ```

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -193,6 +193,19 @@
...
+        <groupId>org.jenkins-ci.tools</groupId>
+        <artifactId>maven-hpi-plugin</artifactId>
+        <configuration>
+          <!-- JENKINS-47074: Build ID precedes display names. -->
+          <compatibleSinceVersion>1.40</compatibleSinceVersion>
+        </configuration>
+      </plugin>
+    </plugins>
+  </build>
+
```
Comment (ikedam) [inline pom.xml]: This change may cause regressions if `yyyy-MM-dd_HH-mm-ss` is used for display names.
(I don't think it's often the case, though).

**Comment by jglick** [comment]
Nonnumeric IDs do not exist after, IIRC, 1.597 or something, which is older than your baseline. So it would suffice to just accept `String` in place of `int`.

**Comment by ikedam** [comment]
I didn’t know that.
So copyartifact already provides a sufficient feature.
I’ll add a test to see that `build.id` can replace `”${build.number}”`.

**Comment by jglick** [comment]
(adding label based on last comment; adjust if I misunderstood)

**Comment by ikedam** [comment]
I added a test to verify that `SpecificBuildSelector` accepts build IDs in #102 .
We don't need to make changes for [JENKINS-47074](https://issues.jenkins-ci.org/browse/JENKINS-47074), and I'll make a new release 1.39.1 only with #96 in the next week.

**Pull request closed without merging**